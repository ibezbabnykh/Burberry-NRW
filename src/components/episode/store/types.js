export const EPISODE_DETAILS_REQUEST = 'EPISODE/EPISODE_DETAILS_REQUEST';
export const EPISODE_DETAILS_SUCCESS = 'EPISODE/EPISODE_DETAILS_SUCCESS';
export const EPISODE_DETAILS_FAILURE = 'EPISODE/EPISODE_DETAILS_FAILURE';