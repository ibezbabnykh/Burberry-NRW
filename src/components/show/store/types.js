export const SHOW_DETAILS_REQUEST = 'SHOW/SHOW_DETAILS_REQUEST';
export const SHOW_DETAILS_SUCCESS = 'SHOW/SHOW_DETAILS_SUCCESS';
export const SHOW_DETAILS_FAILURE = 'SHOW/SHOW_DETAILS_FAILURE';

export const SHOW_EPISODE_LIST_REQUEST = 'SHOW/SHOW_EPISODE_LIST_REQUEST';
export const SHOW_EPISODE_LIST_SUCCESS = 'SHOW/SHOW_EPISODE_LIST_SUCCESS';
export const SHOW_EPISODE_LIST_FAILURE = 'SHOW/SHOW_EPISODE_LIST_FAILURE';