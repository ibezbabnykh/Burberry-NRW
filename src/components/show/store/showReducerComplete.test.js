import reducer from './showReducer';
import * as types from './types';

const initialState = {
  details: {
    data: null,
    error: null,
    loading: false
  },
  episodeList: {
    data: null,
    error: null,
    loading: false
  }
};

describe('show reducer', () => {
  describe('show details reducer', () => {
    it('should handle SHOW_DETAILS_REQUEST and return state', () => {
      const requestAction = {
        type: types.SHOW_DETAILS_REQUEST,
        payload: 1
      }

      const result = reducer(initialState, requestAction);
      expect(result).toMatchSnapshot();
    });

    it('should handle SHOW_DETAILS_SUCCESS and return state', () => {
      const mockResponse = {
        image: {
          medium: 'https://via.placeholder.com/250x140',
          original: 'https://via.placeholder.com/1280x720'
        },
        name: 'show name',
        summary: 'show summary'
      }
      const successAction = {
        type: types.SHOW_DETAILS_SUCCESS,
        payload: mockResponse
      }

      const result = reducer(initialState, successAction);
      expect(result).toMatchSnapshot();
    });

    it('should handle SHOW_DETAILS_FAILURE and return state', () => {
      const mockError = 'Something went wrong';
      const failureAction = {
        type: types.SHOW_DETAILS_FAILURE,
        payload: mockError
      }

      const result = reducer(initialState, failureAction);
      expect(result).toMatchSnapshot();
    });
  });

  describe('episode list reducer', () => {
    it('should handle SHOW_EPISODE_LIST_REQUEST and return state', () => {
      const requestAction = {
        type: types.SHOW_EPISODE_LIST_REQUEST,
        payload: 1
      }

      const result = reducer(initialState, requestAction);
      expect(result).toMatchSnapshot();
    });

    it('should handle SHOW_EPISODE_LIST_SUCCESS and return state', () => {
      const mockResponse = [
        {
          season: '1',
          values: [
            { id: 1, number: 1, season: 1, name: 'episode 1x1', airdate: '2016-04-11' },
            { id: 2, number: 2, season: 1, name: 'episode 1x2', airdate: '2016-04-15' }
          ]
        },
        {
          season: '2',
          values: [
            { id: 3, number: 1, season: 2, name: 'episode 2x1', airdate: '2017-04-11' },
            { id: 4, number: 2, season: 2, name: 'episode 2x2', airdate: '2017-04-15' }
          ]
        }
      ];
      const successAction = {
        type: types.SHOW_EPISODE_LIST_SUCCESS,
        payload: mockResponse
      }

      const result = reducer(initialState, successAction);
      expect(result).toMatchSnapshot();
    });

    it('should handle SHOW_EPISODE_LIST_FAILURE and return state', () => {
      const mockError = 'Something went wrong';
      const failureAction = {
        type: types.SHOW_EPISODE_LIST_FAILURE,
        payload: mockError
      }

      const result = reducer(initialState, failureAction);
      expect(result).toMatchSnapshot();
    });
  });
});