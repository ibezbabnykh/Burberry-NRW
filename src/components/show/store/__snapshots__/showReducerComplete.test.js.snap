// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show reducer episode list reducer should handle SHOW_EPISODE_LIST_FAILURE and return state 1`] = `
Object {
  "details": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
  "episodeList": Object {
    "data": null,
    "error": "Something went wrong",
    "loading": false,
  },
}
`;

exports[`show reducer episode list reducer should handle SHOW_EPISODE_LIST_REQUEST and return state 1`] = `
Object {
  "details": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
  "episodeList": Object {
    "data": null,
    "error": null,
    "loading": true,
  },
}
`;

exports[`show reducer episode list reducer should handle SHOW_EPISODE_LIST_SUCCESS and return state 1`] = `
Object {
  "details": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
  "episodeList": Object {
    "data": Object {
      "data": Array [
        Object {
          "season": "1",
          "values": Array [
            Object {
              "airdate": "2016-04-11",
              "id": 1,
              "name": "episode 1x1",
              "number": 1,
              "season": 1,
            },
            Object {
              "airdate": "2016-04-15",
              "id": 2,
              "name": "episode 1x2",
              "number": 2,
              "season": 1,
            },
          ],
        },
        Object {
          "season": "2",
          "values": Array [
            Object {
              "airdate": "2017-04-11",
              "id": 3,
              "name": "episode 2x1",
              "number": 1,
              "season": 2,
            },
            Object {
              "airdate": "2017-04-15",
              "id": 4,
              "name": "episode 2x2",
              "number": 2,
              "season": 2,
            },
          ],
        },
      ],
    },
    "error": null,
    "loading": false,
  },
}
`;

exports[`show reducer show details reducer should handle SHOW_DETAILS_FAILURE and return state 1`] = `
Object {
  "details": Object {
    "data": null,
    "error": "Something went wrong",
    "loading": false,
  },
  "episodeList": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
}
`;

exports[`show reducer show details reducer should handle SHOW_DETAILS_REQUEST and return state 1`] = `
Object {
  "details": Object {
    "data": null,
    "error": null,
    "loading": true,
  },
  "episodeList": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
}
`;

exports[`show reducer show details reducer should handle SHOW_DETAILS_SUCCESS and return state 1`] = `
Object {
  "details": Object {
    "data": Object {
      "image": Object {
        "medium": "https://via.placeholder.com/250x140",
        "original": "https://via.placeholder.com/1280x720",
      },
      "name": "show name",
      "summary": "show summary",
    },
    "error": null,
    "loading": false,
  },
  "episodeList": Object {
    "data": null,
    "error": null,
    "loading": false,
  },
}
`;
